pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ActivityAnswerCopierSql.java

```diff
@@ -0,0 +1,405 @@
+            }
+        }
+
+        return (T)answerCopier.copyAnswer(getHandle(),
+                sourceQuestionStableId,
+                destinationQuestionStableId,
+                sourceInstanceId,
+                destinationInstanceId,
+                createdAtEpochMillis,
+                lastUpdatedAtEpochMillis, compositeCopyConfiguration);
+    }
+
+    class AnswerToCopy {
+
+        private long answerId;
+
+        private QuestionType questionType;
+
+        @JdbiConstructor
+        public AnswerToCopy(@ColumnName("answer_id") long answerId,
+                            @ColumnName("question_type_code") QuestionType questionType) {
+            this.answerId = answerId;
+            this.questionType = questionType;
+        }
+
+        public long getAnswerId() {
+            return answerId;
+        }
+
+        public QuestionType getQuestionType() {
+            return questionType;
+        }
+    }
+
+    /**
+     * An output from copying an answer, which includes
+     * the source answer id and the destination id at a minimum.
+     */
+    class AnswerCopyResult {
+
+        private final long sourceAnswerId;
+
+        private final long destinationAnswerId;
+
+        public AnswerCopyResult(long sourceAnswerId, long destinationAnswerId) {
+            this.sourceAnswerId = sourceAnswerId;
+            this.destinationAnswerId = destinationAnswerId;
+        }
+
+        public long getSourceAnswerId() {
+            return sourceAnswerId;
+        }
+
+        public long getDestinationAnswerId() {
+            return destinationAnswerId;
+        }
+    }
+
+    interface AnswerCopier<T extends AnswerCopyResult> {
+
+        /**
+         * Copies the answer to one question in an activity instance
+         * to a potentially different question in a different activity instance.
+         */
+        T copyAnswer(Handle handle,
+                     String sourceQuestionStableId,
+                    String destinationQuestionStableId,
+                    long sourceInstanceId,
+                                    long destinationInstanceId,
+                                    long createdAtEpochMillis,
+                                    long lastUpdatedAtEpochMillis,
+                     CompositeAnswerCopyConfiguration compositeCopyConfiguration);
+    }
+
+    /**
+     * Copies top-level answer and composite_answer_item (when directed by compositeCopyConfig)
+     */
+    abstract class BaseAnswerCopier<T extends AnswerCopyResult> implements AnswerCopier {
+
+
+        public T copyAnswer(Handle handle, String sourceQuestionStableId, String destinationQuestionStableId,
+                                           long sourceInstanceId, long destinationInstanceId, long createdAtEpochMillis,
+                            long lastUpdatedAtEpochMillis, CompositeAnswerCopyConfiguration compositeCopyConfig) {
```

Comment:
Doesn't look like we use argument `compositeCopyConfig`

Reply:
A subjective style preference: the argument pattern for a bunch of these methods appears to be:
- sourceStableId
- destinationStableId
- sourceInstanceId 
- destinationInstanceId
Since in my mind it is the combination of actitivty instance id and question stable id that define what a source and destination are, I would find more comforting to see the source specific args next to each other and dest specific args next to each other.

Reply:
Will do!  I actually went back and forth on this one: "should I make each src/dest pair closer or put all sources and all destinations closer"?

Reply:
@MocanaAtBroad it's true that `BaseAnswerCopier` doesn't use `compositeCopyConfig`, but it's here as per the signature of `AnswerCopier`.  We could set things up so that `BaseAnswerCopier` doesn't implement `AnswerCopier` so it's more of a helper that implementing classes can use directly instead of via `super`.  What's your preference?

Reply:
Is there any more utility to `BaseAnswerCopier`? Since it only calls one method `copyBaseAnswer()` I feel like we should just get rid of the inheritance and have the various copiers just do the base copy.